### PR TITLE
[URG ] Angela fixes + converter test fix

### PR DESCRIPTION
**Pull request by mathieucarbou**
- use Angela 3.0.25 (fixes PortManager introduction) - DO NOT USE v3.0.24
- Close NomadClient 
- Fix convertor code test

### CODE REVIEW

**Comment by mathieucarbou** [comment]
Note: release of v3.0.25 is in progress...

File: dynamic-config/server/configuration-provider/src/main/java/org/terracotta/dynamic_config/server/configuration/startup/ConfigurationGeneratorVisitor.java [hunk 1]
```diff
@@ -213,10 +213,11 @@ Path getOrDefaultConfigurationDirectory(String configPath) {
   private void runNomadActivation(Cluster cluster, Node node, NomadServerManager nomadServerManager, Path nodeConfigurationDir) {
     requireNonNull(nodeConfigurationDir);
     NomadEnvironment environment = new NomadEnvironment();
-    NomadClient<NodeContext> nomadClient = new NomadClient<>(singletonList(new NomadEndpoint<>(node.getNodeAddress(), nomadServerManager.getNomadServer())), environment.getHost(), environment.getUser(), Clock.systemUTC());
-    NomadFailureReceiver<NodeContext> failureRecorder = new NomadFailureReceiver<>();
-    nomadClient.tryApplyChange(failureRecorder, new ClusterActivationNomadChange(cluster));
-    failureRecorder.reThrow();
+    try (NomadClient<NodeContext> nomadClient = new NomadClient<>(singletonList(new NomadEndpoint<>(node.getNodeAddress(), nomadServerManager.getNomadServer())), environment.getHost(), environment.getUser(), Clock.systemUTC())) {
```
Comment (saurabhagas) [inline dynamic-config/server/configuration-provider/src/main/java/org/terracotta/dynamic_config/server/configuration/startup/ConfigurationGeneratorVisitor.java:216]: @mathieucarbou this was another place where `NomadClient` wasn't closed.

File: dynamic-config/testing/support/src/main/java/org/terracotta/dynamic_config/test_support/DynamicConfigIT.java [hunk 2]
```diff
@@ -323,16 +323,7 @@ protected Path copyConfigProperty(String configFile) {
...
-    LOGGER.debug("Generating cluster node configuration directories into: {}", configDirs);
-    fn.accept(clusterGenerator);
-    org.terracotta.utilities.io.Files.copy(configDirs.resolve("stripe-" + stripeId).resolve("node-" + nodeId), nodeConfigurationDir, RECURSIVE);
-    LOGGER.debug("Created node configuration directory into: {}", nodeConfigurationDir);
-    return nodeConfigurationDir;
-  }
-
-  protected ConfigurationGenerator getConfigGenerator(Path configDirs) {
-    return new ConfigurationGenerator(configDirs, new ConfigurationGenerator.PortSupplier() {
+    ConfigurationGenerator clusterGenerator = new ConfigurationGenerator(configDirs, new ConfigurationGenerator.PortSupplier() {
```
Comment (saurabhagas) [inline dynamic-config/testing/support/src/main/java/org/terracotta/dynamic_config/test_support/DynamicConfigIT.java:326]: This change was done for this test - with the latest changes, this change wasn't needed, so reverted it.

File: dynamic-config/testing/support/src/main/java/org/terracotta/dynamic_config/test_support/DynamicConfigIT.java [hunk 3]
```diff
@@ -343,6 +334,11 @@ public int getNodeGroupPort(int stripeId, int nodeId) {
         return angela.getNodeGroupPort(stripeId, nodeId);
       }
     });
+    LOGGER.debug("Generating cluster node configuration directories into: {}", configDirs);
```
Comment (saurabhagas) [inline dynamic-config/testing/support/src/main/java/org/terracotta/dynamic_config/test_support/DynamicConfigIT.java:337]: This change was done for this test - with the latest changes, this change wasn't needed, so reverted it.

File: dynamic-config/testing/support/src/main/java/org/terracotta/dynamic_config/test_support/util/ConfigurationGenerator.java [hunk 4]
```diff
@@ -88,7 +88,7 @@ public void generate1Stripe1NodeAndSkipCommit() {
     convert(true, substituteParams(1, 1, "/tc-configs/stripe1-1-node.xml"));
   }
 
-  public Path substituteParams(int stripeId, int nodes, String path) {
+  private Path substituteParams(int stripeId, int nodes, String path) {
```
Comment (saurabhagas) [inline dynamic-config/testing/support/src/main/java/org/terracotta/dynamic_config/test_support/util/ConfigurationGenerator.java:91]: This change was done for this test - with the latest changes, this change wasn't needed, so reverted it.

File: dynamic-config/testing/system-tests/pom.xml [hunk 5]
```diff
@@ -81,6 +81,11 @@
       <artifactId>voter</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
```
Comment (saurabhagas) [inline dynamic-config/testing/system-tests/pom.xml:84]: Needed for `AnyProcess`.

**Pull request merged**